### PR TITLE
Safely add nullable UsedBytes fields

### PR DESCRIPTION
**Pull request by jkohen**
We are seeing crashes in production on this line. UsedBytes is nullable/optional, and kubelet, the code that generates it, always checks for nullability before dereferencing it, so we should do the same here.

### CODE REVIEW

File: metrics/sources/summary/summary.go [hunk 1]
```diff
@@ -273,8 +273,20 @@ func (this *summaryMetricsSource) decodeEphemeralStorageStatsForContainer(metric
...
 	}
-	usage := *rootfs.UsedBytes + *logs.UsedBytes
-	this.addIntMetric(metrics, &MetricEphemeralStorageUsage, &usage)
+	usage := addUsage(rootfs.UsedBytes, logs.UsedBytes)
+	if usage != nil {
+		this.addIntMetric(metrics, &MetricEphemeralStorageUsage, usage)
+	}
+}
+
+func addUsage(first, second *uint64) *uint64 {
```
Comment (igorpeshansky) [inline metrics/sources/summary/summary.go:281]: Would `safeSum` be a better name? This has nothing to do with "usage", per se...

File: metrics/sources/summary/summary.go [hunk 2]
```diff
@@ -273,8 +273,20 @@ func (this *summaryMetricsSource) decodeEphemeralStorageStatsForContainer(metric
 		glog.V(9).Infof("missing storage usage metric!")
 		return
 	}
-	usage := *rootfs.UsedBytes + *logs.UsedBytes
-	this.addIntMetric(metrics, &MetricEphemeralStorageUsage, &usage)
+	usage := addUsage(rootfs.UsedBytes, logs.UsedBytes)
```
Comment (igorpeshansky) [inline metrics/sources/summary/summary.go]: How about:
```go
	if usage := addUsage(rootfs.UsedBytes, logs.UsedBytes); usage != nil {
		this.addIntMetric(metrics, &MetricEphemeralStorageUsage, usage)
	}
```

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: metrics/sources/summary/summary.go (hunk 1 above)
Comment (jkohen) [reply metrics/sources/summary/summary.go:281]: addUsage it's what is used in the kubelet file I copied this from. In any case, you're right it doesn't only work for usage, but I don't know that it generally applies for every sum.

File: metrics/sources/summary/summary.go (hunk 2 above)
Comment (jkohen) [reply metrics/sources/summary/summary.go]: Done.

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

File: metrics/sources/summary/summary.go [hunk 3]
```diff
@@ -19,17 +19,17 @@ import (
 	"net/url"
 	"time"
 
-	. "k8s.io/heapster/metrics/core"
-	"k8s.io/heapster/metrics/sources/kubelet"
+	. "github.com/Stackdriver/heapster/metrics/core"
```
Comment (igorpeshansky) [inline metrics/sources/summary/summary.go:22]: Nice!

**Review by bmoyles0117 — APPROVED** [review]

**Pull request merged**